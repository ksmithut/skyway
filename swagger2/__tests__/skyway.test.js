/* eslint-env jest */
'use strict'

const path = require('path')
const skyway = require('../')

describe('skyway()', () => {
  const fixtures = path.resolve.bind(path, __dirname, 'fixtures')

  test('returns a promise that resolves with the full documentation', () => {
    const api = skyway(fixtures('full.yaml'))
    return expect(api).resolves.toMatchObject({
      swagger: '2.0',
      info: {
        title: 'Full Example',
        version: '1.0.0'
      }
    })
  })

  test('returns rejected promise if there is an error in the swagger docs', () => {
    const api = skyway(fixtures('bad.yaml'))
    return expect(api).rejects.toMatchSnapshot()
  })

  test('accepts validation options', () => {
    const api = skyway(fixtures('bad.yaml'), {
      validate: { schema: false, spec: false }
    })
    return expect(api).resolves.toMatchObject({
      foo: 'bar',
      swagger: '2.0',
      info: {
        title: 'Full Example',
        version: '1.0.0'
      }
    })
  })
})
