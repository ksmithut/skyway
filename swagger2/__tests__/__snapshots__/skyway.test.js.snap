// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`skyway() returns rejected promise if there is an error in the swagger docs 1`] = `
[SyntaxError: Swagger schema validation failed. 
  Additional properties not allowed: foo at #/
 
JSON_OBJECT_VALIDATION_FAILED]
`;
